### PR TITLE
feat: integreate lndhub create wallet

### PR DESCRIPTION
**Pull request by musayann**

### CODE REVIEW

File: admin/connections.php [hunk 1]
```diff
@@ -154,10 +181,17 @@ public function field_lndhub_login()
...
+            '<input id="lndhub_password" type="password" name="%s" value="%s" autocomplete="off" /><br><label>%s</label>',
             $this->get_field_name('lndhub_password'),
             $this->get_field_value('lndhub_password'),
             'LndHub Password'
         );
     }
+    public function field_lndhub_generate()
+    {
+        printf(
+            '<button id="lndhub_create_account" class="button button-primary" type="button">%s</button>',
```
Comment (bumi) [inline admin/connections.php:193]: let's show this button only if no connection is configured yet.

File: admin/connections.php [hunk 2]
```diff
@@ -52,21 +53,47 @@ public function renderer()
...
+                            if (!lndConnectUrl) {
+                                return;
+                            }
+                            var url = new URL(lndConnectUrl);
+                            document.getElementById('lnp_lnd_address').value = 'https:' + url.pathname;
+                            document.getElementById('lnp_lnd_macaroon').value = url.searchParams.get('macaroon');
+                            document.getElementById('lnp_lnd_cert').value = url.searchParams.get('cert');
+                        });
+
+                        document.getElementById('lndhub_create_account').addEventListener('click', function(e) {
```
Comment (bumi) [inline admin/connections.php:71]: I think you are missing a `e.preventDefault()` here?

**Review by bumi — CHANGES_REQUESTED** [review]
I currently get redirected to options.pho and see a JSON response. - I haven't tested the form in a while, likely this was before this PR already, but can you have look.
Does it work for you?

File: admin/connections.php (hunk 2 above)
Comment (musayann) [reply admin/connections.php:75]: Great catch thanks. just added it

**Comment by musayann** [comment]
> I currently get redirected to options.pho and see a JSON response. - I haven't tested the form in a while, likely this was before this PR already, but can you have look. Does it work for you?

Took a look and the settings page seems to work well. Tried with a fresh installation of WordPress. not sure if the bug shows up with an old installation of WordPress containing previous versions of the app. perhaps if you have such a database you can share it just so I can do more testing.

**Pull request merged**